src/components/core/GeoChart.tsx

```diff
@@ -0,0 +1,239 @@
+// (C) 2007-2019 GoodData Corporation
+import * as React from "react";
+import isEqual = require("lodash/isEqual");
+import noop = require("lodash/noop");
+import mapboxgl = require("mapbox-gl");
+import { WrappedComponentProps } from "react-intl";
+
+import { ICommonChartProps } from "./base/BaseChart";
+import { BaseVisualization } from "./base/BaseVisualization";
+import {
+    commonDefaultProps,
+    ILoadingInjectedProps,
+    visualizationLoadingHOC,
+} from "./base/VisualizationLoadingHOC";
+import { createDataSource, IGeoDataSource, IGeoDataSourceFeatures } from "./geoChart/dataSource";
+
+import { IDataSourceProviderInjectedProps } from "../afm/DataSourceProvider";
+import { LoadingComponent } from "../simple/LoadingComponent";
+
+import "../../../styles/css/geoChart.css";
+
+const DEFAULT_LOCATION = {
+    LAT: 34,
+    LNG: 5,
+    ZOOM: 2,
+};
+
+const MAP_PIN_COLORS = [
+    "#fbb03b",
+    "#223b53",
+    "#e26a6a",
+    "#aa8f00",
+    "#3498db",
+    "#3477db",
+    "#2277db",
+    "#bb77db",
+    "#0077db",
+    "#3aa7db",
+];
+const MAP_PIN_SIZES = [8, 9, 12, 16, 25];
+
+const MAPBOX_ACCESS_TOKEN =
+    "pk.eyJ1IjoiaW1udXR6IiwiYSI6ImNrMHAxY2UxZzBnc2EzZG11YmVhd2dubG0ifQ.bUTN7ceAHq6kVooe3MKgqg";
+const MAPBOX_STYLE = "mapbox://styles/mapbox/streets-v11";
+mapboxgl.accessToken = MAPBOX_ACCESS_TOKEN;
```

Comment:
we should move these constants to new `/gooddata-react-components/src/constants/geoChart.ts`
and add type for them, like
```
...
const MAP_PIN_COLORS: string[];
const MAP_PIN_SIZES: number[];
...
```

Reply:
SD-780 will handle this